### PR TITLE
avoid report logs in sentry

### LINKED ISSUE #329: avoid  report messages from getting to sentry
probably need something in hijack loggers to prevent those messages (below exception) from getting to sentry

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #329 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

and add a comment to request a report recompute if it isn't computed in 30 minutes for #526

### CODE REVIEW

File: solarforecastarbiter/reports/templates/html/pending.html [hunk 1]
```diff
@@ -5,6 +5,7 @@ <h1 id="report-title">{{ report.report_parameters.name }}</h1>
 {% endblock %}
 <div class="alert alert-info">
   Report generation is pending. We will try and fetch the report again in 30 seconds, or you can try reloading this page.
+  If no report is available after 30 minutes, try recomputing the report.
```
Comment (wholmgren) [inline solarforecastarbiter/reports/templates/html/pending.html:8]: do you have benchmark to suggest that 30 minutes is the right number? in my experience it works in less 5 minutes or not at all.

File: solarforecastarbiter/utils.py [hunk 2]
```diff
@@ -268,10 +269,15 @@ def hijack_loggers(loggers, level=logging.INFO):
         logger = logging.getLogger(name)
         old_handlers[name] = logger.handlers
         logger.handlers = [handler]
+        sentry_logging.ignore_logger(name)
     yield handler
     for name in loggers:
         logger = logging.getLogger(name)
         logger.handlers = old_handlers[name]
+        try:
+            sentry_logging._IGNORED_LOGGERS.remove(name)
```
Comment (wholmgren) [inline solarforecastarbiter/utils.py:278]: use of private attribute is worrisome, but I don't really understand what's happening here. I assume you considered an alternative so ok to move ahead.

**Review by wholmgren — APPROVED** [review]
I don't fully understand the issue. Is this about proprietary data?

File: solarforecastarbiter/reports/templates/html/pending.html (hunk 1 above)
Comment (alorenzo175) [reply solarforecastarbiter/reports/templates/html/pending.html:8]: Nope. I just picked a number that I thought would definitely be long enough considering possibly more congestion than there is now

File: solarforecastarbiter/utils.py (hunk 2 above)
Comment (alorenzo175) [reply solarforecastarbiter/utils.py:278]: yeah there isn't a good alternative considering that this code is used by the api and doesn't access the sentry init where other tricks could be used. At least there is a test that will fail if this changes

**Comment by alorenzo175** [comment]
> I don't fully understand the issue. Is this about proprietary data?

Somewhat. Data probably wouldn't be logged, but right now things like report failure due to no data are making it to sentry with the forecast/observations names. Best if we don't even see that info

**Pull request merged**